server/src/operators/dataset_operator.rs

```diff
@@ -274,18 +277,67 @@ pub async fn clear_dataset_by_dataset_id_query(
     Ok(())
 }
 
-pub async fn delete_chunks_in_dataset(
+pub async fn delete_items_in_dataset(
     id: uuid::Uuid,
+    deleted_at: chrono::NaiveDateTime,
     pool: web::Data<Pool>,
     clickhouse_client: web::Data<clickhouse::Client>,
     dataset_config: DatasetConfiguration,
 ) -> Result<(), ServiceError> {
+    use crate::data::schema::chunk_group::dsl as chunk_group;
+    use crate::data::schema::chunk_group_bookmarks::dsl as chunk_group_bookmarks_columns;
     use crate::data::schema::chunk_metadata::dsl as chunk_metadata_columns;
+    use crate::data::schema::files::dsl as files_column;
 
     let mut conn = pool.get().await.unwrap();
 
     let qdrant_collection = format!("{}_vectors", dataset_config.EMBEDDING_SIZE);
 
+    let chunk_groups = chunk_group::chunk_group
+        .filter(chunk_group::dataset_id.eq(id))
+        .select(chunk_group::id)
+        .load::<uuid::Uuid>(&mut conn)
+        .await
+        .map_err(|err| {
+            log::error!("Could not fetch groups: {}", err);
+            ServiceError::BadRequest("Could not fetch groups".to_string())
+        })?;
+
+    diesel::delete(
+        chunk_group_bookmarks_columns::chunk_group_bookmarks
+            .filter(chunk_group_bookmarks_columns::group_id.eq_any(chunk_groups)),
+    )
+    .execute(&mut conn)
+    .await
+    .map_err(|err| {
+        log::error!("Could not delete groups: {}", err);
+        ServiceError::BadRequest("Could not delete groups".to_string())
+    })?;
+
+    diesel::delete(
+        chunk_group::chunk_group
+            .filter(chunk_group::dataset_id.eq(id))
+            .filter(chunk_group::created_at.lt(deleted_at)),
+    )
+    .execute(&mut conn)
+    .await
+    .map_err(|err| {
+        log::error!("Could not delete groups: {}", err);
+        ServiceError::BadRequest("Could not delete groups".to_string())
+    })?;
+
+    diesel::delete(
+        files_column::files
+            .filter(files_column::dataset_id.eq(id))
+            .filter(files_column::created_at.lt(deleted_at)),
+    )
+    .execute(&mut conn)
+    .await
+    .map_err(|err| {
+        log::error!("Could not delete groups: {}", err);
+        ServiceError::BadRequest("Could not delete groups".to_string())
+    })?;
+
```

Comment:
I worry about this for datasets which have 10k+ groups, but idk. I think it's fine for now, but something to think about if we see usage on these resources start to spike.